### ISSUE TITLE
[Bug] Duplicate functions when loading an architecture plugin for an existing architecture

### ISSUE THREAD

**Issue by Arcnor** [Vector35/binaryninja-api#1150]
I'm trying to fix #1140 by creating an `ArchitectureHook` plugin. To start, I've been trying to create a no-op hook to verify that everything works as expected:

```
from binaryninja.architecture import Architecture, ArchitectureHook

class MIPSEL32Fix(ArchitectureHook):
    def get_instruction_low_level_il(self, data, addr, il):
        return super(MIPSEL32Fix, self).get_instruction_low_level_il(data, addr, il)

# Install the hook by constructing it with the desired architecture to hook, then registering it
MIPSEL32Fix(Architecture['mipsel32']).register()
```

However, when opening an existing DB all functions get duplicated. I've also tried to return `0` or `None` in case that will work, but I get the same result.

**Comment by Arcnor** [comment]
Ok, it seems that this doesn't happen to *every* function, but to 90-95% of them. I have no idea what the criteria is, but some random ones are not duplicated.

**Comment by psifertex** [comment]
Possibly ones that were new as a result of your changes?

**Comment by psifertex** [comment]
Oh, nm, just realized a NO-OP layer shouldn't change any results. :-)

**Comment by rssor** [comment]
Should be resolved in `1.1.1494-dev`.

**Comment by Arcnor** [comment]
Will try this soon, thanks!
